### PR TITLE
Feature/massban

### PR DESCRIPTION
**Pull request by McJeffr**

### CODE REVIEW

File: src/Command/Massban.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
+const Command = require("../Command");
+const RoleDeterminer = require("../Helper/RoleDeterminer");
+const writeToLog = require("../Helper/GenericLogger");
```
Comment (navarr) [inline src/Command/Massban.js]: This should be part of the dependency graph, not an in-file requirement.

Reason being: to make it mockable for unit testing

**Review by navarr — CHANGES_REQUESTED** [review]
I have concerns with this command - chiefly the amount of false positives it can capture

File: src/Command/Massban.js (hunk 1 above)
Comment (McJeffr) [reply src/Command/Massban.js]: None of the helpers in any of the commands are part of the dependency graph, but I guess I can add these ones

**Comment by McJeffr** [comment]
> I have concerns with this command - chiefly the amount of false positives it can capture

That is true, but I do not expect us to use this command every day. Furthermore, the command is restricted to a maximum interval of 5 minutes, and all of the bots join in one large swath within 1 minute. There is a risk of one (or maybe 2) regular members joining in this same interval, but that's just tough luck. They can appeal their ban if they want to. Realistically speaking, I don't think it will catch many false positives, if any at all.

**Review by navarr — APPROVED** [review]

**Pull request merged**